pkg/udevevent/monitor.go

```diff
@@ -98,12 +117,35 @@ func (m *monitor) process(fd int) error {
 		device.UdevDeviceUnref()
 		return nil
 	}
-	event := newEvent()
-	event.process(device)
-	event.send()
+	var eventType UdevEventType
+	switch device.GetAction() {
+	case libudevwrapper.UDEV_ACTION_ADD:
+		eventType = EventTypeAdd
+	case libudevwrapper.UDEV_ACTION_REMOVE:
+		eventType = EventTypeRemove
+	default:
+		eventType = EventTypeChange
+	}
+	event := UdevEvent{device, eventType}
+	dispatchEvent(event)
 	return nil
 }
 
+func dispatchEvent(event UdevEvent) {
+	for _, sub := range subscriptions {
```

Comment:
Is this created so that there can be multiple subscribers for the events?

Reply:
yes